app/code/core/Mage/GoogleAnalytics/Block/Ga.php

```diff
@@ -333,41 +341,75 @@ protected function _getOrdersTrackingCodeAnalytics4()
             $eventData['items'] = [];
 
             foreach ($productCollection as $productInCart) {
+                $_product = Mage::getModel('catalog/product')->load($productInCart->getProductId());
                 $eventData['items'][] = [
-                    'id' => $productInCart->getSku(),
-                    'name' => $productInCart->getName(),
-                    'list_name' => 'Product Detail Page',
-                    'brand' => $productInCart->getAttributeText('manufacturer'),
-                    'category' => 'Products',
-                    'price' => number_format($productInCart->getFinalPrice(), 2),
+                    'item_id' => $_product->getSku(),
+                    'item_name' => $_product->getName(),
+                    'item_brand' => $_product->getAttributeText('manufacturer'),
+                    'price' => number_format($_product->getFinalPrice(), 2),
                 ];
-                $eventData['value'] += $productInCart->getFinalPrice();
+                $eventData['value'] += $_product->getFinalPrice();
             }
             $eventData['value'] = number_format($eventData['value'], 2);
             $result[] = "gtag('event', 'view_cart', " . json_encode($eventData, JSON_THROW_ON_ERROR) . ");";
         }
 
         //begin checkout
-        elseif ($moduleName == 'checkout' && $controllerName == 'onepage') {
+        elseif (($moduleName == 'checkout' && $controllerName == 'onepage') || ($moduleName == 'firecheckout' && $controllerName == 'index')) {
```

Comment:
this worries me a bit, we should have a configurable something in order to support all of the checkout systems out there. if we hardcode "checkout/onepage" is something, but I feel weird hardcoding the name of just one 3rd party checkout